Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -253,6 +212,65 @@ - (void)urlSessionTaskResume:(NSURLSessionTask *)sessionTask
     }
 }
 
+- (void)addTraceWithoutTransactionToTask:(NSURLSessionTask *)sessionTask
+{
+    SentryPropagationContext *propagationContext = SentrySDK.currentHub.scope.propagationContext;
+    [self addBaggageHeader:[[propagationContext traceContext] toBaggage]
+               traceHeader:[propagationContext traceHeader]
+                 toRequest:sessionTask];
+}
+
+- (void)addBaggageHeader:(SentryBaggage *)baggage
+             traceHeader:(SentryTraceHeader *)traceHeader
+               toRequest:(NSURLSessionTask *)sessionTask
+{
+    if ([self sessionTaskRequiresPropagation:sessionTask]) {
+        NSString *baggageHeader = @"";
+
+        if (baggage != nil) {
+            baggageHeader =
+                [baggage toHTTPHeaderWithOriginalBaggage:
+                             [SentrySerialization
+                                 decodeBaggage:sessionTask.currentRequest
+                                                   .allHTTPHeaderFields[SENTRY_BAGGAGE_HEADER]]];
+        }
+
+        // First we check if the current request is mutable, so we could easily add a new
+        // header. Otherwise we try to change the current request for a new one with the extra
+        // header.
+        if ([sessionTask.currentRequest isKindOfClass:[NSMutableURLRequest class]]) {
+            NSMutableURLRequest *currentRequest = (NSMutableURLRequest *)sessionTask.currentRequest;
+            [currentRequest setValue:traceHeader.value forHTTPHeaderField:SENTRY_TRACE_HEADER];
+
+            if (baggageHeader.length > 0) {
+                [currentRequest setValue:baggageHeader forHTTPHeaderField:SENTRY_BAGGAGE_HEADER];
+            }
+        } else {
+            // Even though NSURLSessionTask doesn't have 'setCurrentRequest', some subclasses
+            // do. For those subclasses we replace the currentRequest with a mutable one with
+            // the additional trace header. Since NSURLSessionTask is a public class and can be
+            // override, we believe this is not considered a private api.
```

Comment:
This is interesting. What subclasses are we talking about here? I checked all of the public subclasses and none of them have this method: `NSURLSessionDataTask`, `NSURLSessionUploadTask`, `NSURLSessionDownloadTask`, `NSURLSessionStreamTask` and `NSURLSessionWebSocketTask`.

This doesn't have to be addressed in this PR since it's outside scope, but this is my first time seeing this and it piqued my interest.

Reply:
Apple's own subclasses that are returned from `NSURLSession:startTask`